ethers-etherscan/src/verify.rs

```diff
@@ -24,14 +24,19 @@ pub struct VerifyContract {
     /// applicable when codeformat=solidity-single-file
     #[serde(skip_serializing_if = "Option::is_none")]
     pub runs: Option<String>,
-    /// NOTE: the alias with a typo is due to an old typo in the etherscan API
-    /// `constructorArguements`, but blockscout requires that it's spelled correctly.
-    #[serde(
-        rename = "constructorArguments",
-        alias = "constructorArguements",
-        skip_serializing_if = "Option::is_none"
-    )]
+    /// The constructor arguments for the contract, if any.
+    /// NOTE: This is obviously misspelled. The reason for this is that Etherscan has had
+    /// this misspelling on their API for quite a long time, and changing it would break
+    /// verification with arguments.
+    /// For instances (e.g some blockscout instances) that might support the proper field
+    /// and not the misspelling, the private field `blockscout_constructor_arguments` is
+    /// populated with the exact arguments passed to this field as well.
+    #[serde(rename = "constructorArguements", skip_serializing_if = "Option::is_none")]
     pub constructor_arguments: Option<String>,
+    /// Properly spelled constructor arguments. This is needed as some blockscout instances
+    /// can identify the correct spelling instead of the typo version above.
+    #[serde(rename = "constructorArguments", skip_serializing_if = "Option::is_none")]
+    blockscout_constructor_arguments: Option<String>,
```

Comment:
Use `serde(alias = "...")`

```suggestion
    /// The constructor arguments for the contract, if any.
    ///
    /// NOTE: This is also parsed as the misspelled `constructorArguements`. The reason for this is that Etherscan has had
    /// this misspelling on their API for quite a long time, and changing it would break
    /// verification with arguments. Other services have fixed this typo, so we support both.
    #[serde(rename = "constructorArguments", alias = "constructorArguements", skip_serializing_if = "Option::is_none")]
    pub constructor_arguments: Option<String>,
```

Reply:
Oh yeah, happy to turn this field public—this shouldn't be breaking.

However, we can't use `alias` as this only works for deserialization, not serialization, so we'd end up in the same place as before (not serializing etherscan's `constructorArguements` correctly)

Reply:
Makes sense